### PR TITLE
Update CI to report tests results

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -249,11 +274,21 @@ jobs:
           secret_token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Compile and run tests
-        uses: eProsima/eProsima-CI/multiplatform/colcon_build_test_flaky@v0
+        id: compile_and_test
+        uses: eProsima/eProsima-CI/multiplatform/colcon_build_test@feature/test-results
```
Comment (Tempate) [inline .github/workflows/test.yml:278]: Why not use `colcon_build_test_flaky`?

Comment (rsanchez15) [reply .github/workflows/test.yml:278]: While reviewing the eProsima-CI action I saw that `colcon_build_test_flaky` is exactly the same as `colcon_build_test` but whith come extra `ctest-args`. I'd prefer to use the same for both and add this extra configuration on the workflow

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -107,14 +107,23 @@ jobs:
           secret_token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Compile and run tests
-        uses: eProsima/eProsima-CI/multiplatform/colcon_build_test@v0
+        id: compile_and_test
+        uses: eProsima/eProsima-CI/multiplatform/colcon_build_test@feature/test-results
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply .github/workflows/test.yml]: Addressed in 3c71d88

File: .github/workflows/test.yml (hunk 1 above)
Comment (Tempate) [reply .github/workflows/test.yml]: Okay, we can remove it in the future.

**Review by Tempate — DISMISSED** [review]
LGTM

**Pull request merged**